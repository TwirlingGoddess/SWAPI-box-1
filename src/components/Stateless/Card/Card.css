.card {
  text-align: center;
  box-sizing: border-box;
  max-height: 60vh;
  width: 36vh;
  background-color: #000000e0;
  -webkit-box-shadow: 90px 38px 85px 26px #ff000091;
  box-shadow: 90px 38px 85px 26px #ff000091;
  margin-bottom: 7vh;
  color: white;
  line-height: 9vh; }

.favoriteBar {
  background-image: url("https://www.shareicon.net/data/2015/09/14/100862_star-wars_512x512.png");
  width: 100%;
  height: 6vh;
  background-size: contain;
  background-color: white;
  background-repeat: no-repeat; }

.favoriteCard {
  text-align: center;
  box-sizing: border-box;
  max-height: 60vh;
  width: 36vh;
  background-color: #0400ffe0;
  margin-bottom: 7vh;
  color: white;
  line-height: 9vh; }

.cardText {
  padding: 4vh; }
