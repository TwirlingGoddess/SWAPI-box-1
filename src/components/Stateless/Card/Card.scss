.card {
  text-align: center;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  max-height: 60vh;
  width: 36vh;
  background-color: #000000e0;
  box-shadow: 1px -2px 20px 2px #ff000091;
  margin-bottom: 7vh;
  color: white;
  line-height: 9vh;
  cursor: pointer;
}

.favoriteBar {
  background-image:url('https://www.shareicon.net/data/2015/09/14/100862_star-wars_512x512.png');
  width: 100%;
  background-size: 6vh;
  background-color: #be813c;
  background-repeat: no-repeat;
  max-height: 10vh;
  overflow: scroll;
}

.favoriteCard {
  text-align: center;
  box-sizing: border-box;
  max-height: 60vh;
  width: 36vh;
  margin-bottom: 7vh;
  color: white;
  line-height: 9vh;
  cursor: pointer;
  background: -webkit-linear-gradient(transparent, transparent),
  -webkit-linear-gradient(top, rgba(213,173,109,1) 0%, rgba(213,173,109,1) 26%, rgba(226,186,120,1) 35%, rgba(163,126,67,1) 45%, rgba(145,112,59,1) 61%, rgba(213,173,109,1) 100%);
  background: -o-linear-gradient(transparent, transparent);
  box-shadow: 1px -2px 20px 2px #1602c5ad;
  
}

.cardText {
  padding: 4vh 1vh;
}

.name {
  font-size: 3vh;
  padding: 0px 3.2vh;
  text-shadow: 1px 1px 1px #201609;
}

h2 {
  font-size: 2.5vh;
}

.residentsBox {

}