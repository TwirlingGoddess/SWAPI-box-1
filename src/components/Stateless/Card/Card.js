import './Card.css'
import React from 'react'
import PropTypes from 'prop-types';



const Card = ({data, id, clicked, findCard}) => {
  const card = data
  return (
    <div className={clicked ? 'favoriteCard' : 'card'  } onClick={() => findCard(data)}>
      <div className='favoriteBar'>
        <h2 className='name'>{'Name' + card.Name}</h2>
      </div> 
      <div className='cardText'> 
        <h2>{card.Homeworld}</h2>
        <h2>{card.Specie}</h2>
        <h2>{card.Population}</h2>
        <h2>{card.Model}</h2>
        <h2>{card.Class}</h2>
        <h2>{card.Passengers}</h2>
        <h2>{card.Terrain}</h2>      
        <h2>{card.Climate}</h2>
        <h2>{card.Population}</h2>
        <div className='redidentsBox'>
          <h2 className='residents'>{card.Residents}</h2>
        </div>
      </div>
    </div>
  )
}

Card.propTypes = {
  data: PropTypes.object.isRequired,
  key: PropTypes.number,
  clicked: PropTypes.bool,
  findCard: PropTypes.func.isRequired,
};


export default Card