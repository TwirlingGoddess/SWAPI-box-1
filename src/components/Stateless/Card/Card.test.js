import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card';

let mockProps = {
  data:{},
  id: 2,
  clicked: false,
  findCard: jest.fn()
};

describe('Card', () => {
  let card;
  
  beforeEach(() => {
    card = shallow(<Card {...mockProps} />, { disableLifecycleMethods: true });
  })

  it('matches snapshot', () => {
    expect(card).toMatchSnapshot();
  })

  it("doesn't have a class of 'clicked' if the card has not been selected", () => {
    mockProps.clicked = false;
    expect(card.hasClass('favoriteCard')).toEqual(false);
  });
  it("has a class of 'favoriteCard' if the card has been selected", () => {
    mockProps.clicked = true;
    card = shallow(<Card {...mockProps} />);
  
    expect(card.hasClass('favoriteCard')).toBe(true);
  });

  it('calls findCard when card is clicked', () => {
    let mockProps = {
      data:{},
      id: 2,
      clicked: false,
      findCard: jest.fn()
    };
    card = mount(<Card {...mockProps} />);
    card.find('.card').simulate('click');
    expect(card.prop('findCard')).toHaveBeenCalledTimes(1);

    it('it has a favoriteBar active class if its a favorite card', () => {
      mockProps.clicked= true
      const favBar = card.find('.favoriteActiveBar');
      expect(favBar.hasClass('.favoriteActiveBar')).toBe(true);
    });
  });
})

