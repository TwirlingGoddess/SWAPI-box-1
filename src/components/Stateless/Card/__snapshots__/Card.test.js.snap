// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    clicked={false}
    data={Object {}}
    findCard={[Function]}
    id={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="favoriteBar"
>
          <h2
                    className="name"
          >
                    Name: undefined
          </h2>
</div>,
        <div
          className="cardText"
>
          <div
                    className="redidentsBox"
          >
                    <h2
                              className="residents"
                    />
          </div>
</div>,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="name"
>
            Name: undefined
</h2>,
          "className": "favoriteBar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name: undefined",
            "className": "name",
          },
          "ref": null,
          "rendered": "Name: undefined",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [],
            <div
              className="redidentsBox"
>
              <h2
                            className="residents"
              />
</div>,
          ],
          "className": "cardText",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2
                className="residents"
/>,
              "className": "redidentsBox",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "residents",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="favoriteBar"
>
            <h2
                        className="name"
            >
                        Name: undefined
            </h2>
</div>,
          <div
            className="cardText"
>
            <div
                        className="redidentsBox"
            >
                        <h2
                                    className="residents"
                        />
            </div>
</div>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="name"
>
              Name: undefined
</h2>,
            "className": "favoriteBar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name: undefined",
              "className": "name",
            },
            "ref": null,
            "rendered": "Name: undefined",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [],
              <div
                className="redidentsBox"
>
                <h2
                                className="residents"
                />
</div>,
            ],
            "className": "cardText",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2
                  className="residents"
/>,
                "className": "redidentsBox",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "residents",
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
